src/goto-programs/resolve_inherited_component.h

```diff
@@ -70,7 +70,8 @@ class resolve_inherited_componentt
     const irep_idt &component_name,
     const irep_idt &user_class_name);
 
-  class_hierarchyt class_hierarchy;
+  const class_hierarchyt &class_hierarchy;
+  class_hierarchyt owned_class_hierarchy;
```

Comment:
This was purely to get around the fact that reference fields require some value on construction and that class_hierarchyt gets populated by a () operator, so needs something to exist before passing a symbol table too it.

I originally wanted to keep changes as localised as possible, but I could add a constructor to class_hierarchyt that takes a symbol_tablet and remove the need for this, it would just have to be a single-param implicit one, which I've been told is a bad thing.

Reply:
Hmm, is the constructor that take a single argument (just the symbol table) actually ever called!? Maybe the simple fix is to get rid of that constructor?